v8.go

```diff
@@ -77,14 +77,19 @@ type Snapshot struct{ data C.StartupData }
 
 func newSnapshot(data C.StartupData) *Snapshot {
 	s := &Snapshot{data}
-	runtime.SetFinalizer(s, (*Snapshot).release)
```

Comment:
hrm... actually, all we need to wrap the context one level and put the finalizer on the wrapper.

The map and the values can all point back to the internal wrapper object, and the outer wrapper is only for the external API.  When all references to the external object wrapper are gone, then we can finalize the context.

On the other hand... maybe somebody wants to fire off a long-running operation?  Perhaps the explicit release is better anyways.